lang/en/docs/selective-version-resolutions.md

```diff
@@ -34,12 +34,21 @@ Add a `resolutions` field to your `package.json` file and define your version ov
     "d2": "file:../d2-1"
   },
   "resolutions": {
-    "d2/left-pad": "1.1.1",
-    "c/**/left-pad": "1.1.2"
+    "d2/left-pad": "2.0.0",
+    "c/**/left-pad": "3.0.0",
+    "**/right-pad": "1.0.0"
   }
 }
 ```
 
+In this example, all packages requiring `right-pad` will be given the same version. However, this will not always work for your codebase. Any package that wants to require version 6.0.0, will be forced to require version 1.0.0 and that may result in packages breaking.
+
+Another approach is to scope the resolution to a specific dependency, allowing multiple versions to still be used. In this case, when you import `left-pad` directly, it will be version 1.0.0, but when `d2` imports `left-pad` directly it will get version 2.0.0.
+
+However, if `d2` imports a dependency and that dependency imports `left-pad`, it will not likely be version 2.0.0, as the resolution does not apply in this case. To prevent this from happening, `c` uses `**` to apply the resultion for all of its dependencies as well.
```

Comment:
is `c` the name of a directory here?

Reply:
Yeah those names also confused me a bit, but `c` is defined as a dependency: `"c": "file:../c-1"`

I chose not to change the original example (keeping this PR as small as possible), but it may be clearer to use actual dependency names here. `c` and `d2` could just `jquery` and `lodash`, which would make it easier to read I think.

Reply:
I'd prefer this too, although for the `file` dependency, something like `my-local-dependency` might stop some confusion too

Reply:
Completely re-wrote it. It's a bit lengthy now, but with easier names and easier version numbers, hopefully it's relatively easy to read.

I also removed the file dependencies. I felt that they might just make it more difficult to understand the resolutions, which is what this page is about.

What do you think?